### DISCUSSION TITLE
How To Return All Validation Errors

### DISCUSSION THREAD

**Discussion by MikeO89** [hyperjump-io/json-schema#32] [Q&A] [answered]
Example schema:

`{`
`    "$schema": "https://json-schema.org/draft/2020-12/schema",`
`    "title": "Example Schema.",`
`    "description": "Some example schema.",`
`    "type": "object",`
`    "properties": {`
`        "firstName": {`
`            "$ref": "#/$defs/name"`
`        },`
`        "surname": {`
`            "$ref": "#/$defs/name"`
`        },`
`        "age": {`
`            "type": "number",`
`            "minimum": 0`
`        }`
`    },`
`    "required": [`
`        "firstName",`
`        "surname",`
`        "age"`
`    ],`
`    "$defs": {`
`        "name": {`
`            "type": "string",`
`            "minLength": 3`
`        }`
`    },`
`    "additionalProperties": false`
`}`

Example Node JS Code:

`const person = {`
`  "firstName": "",`
`  "age": "25"`
`};`
`const isValidResult = await validate(url);`
`const result = isValidResult (person, VERBOSE);`

The following 3 errors exist:

- Message: String '' is less than minimum length of 3.
Schema path: #/$defs/name/minLength

- Message: Invalid type. Expected Number but got String.
Schema path: #/properties/age/type

- Message: Required properties are missing from object: surname.
Schema path: #/required

`result.errors` contains 1/3 of the errors and is not reporting 3/3 errors (all three). How would I achieve this, if possible?

**Comment by jdesrosiers** [comment, marked as answer]
This implementation doesn't support returning all errors. It returns immediately when the first error is found.

**Reply by Rinaldo** [reply]
Is returning all errors something you have on your roadmap? I’d like to migrate from avj to this library but the lack of an allErrors flag is a blocker for me

**Reply by jdesrosiers** [reply]
Yes, in fact I was thinking about this just the other day. I can't say when it's likely to happen, but it is something I'm thinking about.
